From db4cf2d5407ad0e67a0693913037ec94d1dc5100 Mon Sep 17 00:00:00 2001
From: Wei Xu <weixugc@google.com>
Date: Wed, 4 Mar 2020 10:27:40 -0800
Subject: [PATCH] Skip applying ssh keys to 'root'.

/root is read-only on COS.  Skip applying the ssh keys to 'root'
to avoid unnecessary error messages.
---
 cloudinit/config/cc_ssh.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cloudinit/config/cc_ssh.py b/cloudinit/config/cc_ssh.py
index f8f7cb35..4ca07c02 100755
--- a/cloudinit/config/cc_ssh.py
+++ b/cloudinit/config/cc_ssh.py
@@ -199,6 +199,8 @@ def apply_credentials(keys, user, disable_root, disable_root_opts):
     if user:
         ssh_util.setup_user_keys(keys, user)
 
+    # Don't apply the ssh keys to 'root' because /root is read-only on COS.
+    """
     if disable_root:
         if not user:
             user = "NONE"
@@ -208,5 +210,6 @@ def apply_credentials(keys, user, disable_root, disable_root_opts):
         key_prefix = ''
 
     ssh_util.setup_user_keys(keys, 'root', options=key_prefix)
+    """
 
 # vi: ts=4 expandtab
-- 
2.25.0.265.gbab2e86ba0-goog

